packages/sdk/src/fetcher/on-chain/index.ts

```diff
@@ -251,38 +264,43 @@ class Fetcher implements IPartialCarrotFetcher {
     }
 
     public async fetchOracles({
-        provider,
+        publicClient,
         addresses,
     }: FetchEntitiesParams): Promise<{ [address: string]: Oracle }> {
-        const { chainId } = await provider.getNetwork();
+        const chainId = await publicClient.getChainId();
         enforce(chainId in ChainId, `unsupported chain with id ${chainId}`);
         const chainAddresses = CHAIN_ADDRESSES[chainId as ChainId];
-        const multicall = new Contract(
-            chainAddresses.multicall,
-            MULTICALL_ABI,
-            provider
-        );
-        const oraclesManagerContract = new Contract(
-            chainAddresses.oraclesManager,
-            ORACLES_MANAGER_ABI,
-            provider
-        );
+        const oraclesManager = getContract({
+            abi: ORACLES_MANAGER_ABI,
+            address: chainAddresses.oraclesManager,
+            publicClient: publicClient,
+        });
 
-        const oracleAmounts = await oraclesManagerContract.templatesAmount();
-        if (oracleAmounts.isZero()) return {};
+        const oracleAmounts = await oraclesManager.read.templatesAmount();
+        if (oracleAmounts == 0n) return {};
         const oracleAddresses =
             addresses && addresses.length > 0
                 ? addresses
-                : await oraclesManagerContract.enumerate(0, oracleAmounts);
+                : (
+                      await oraclesManager.read.enumerate([0n, oracleAmounts])
+                  ).map((oracle) => oracle.addrezz);
 
-        const [, oraclesResult] = await multicall.callStatic.aggregate(
-            oracleAddresses.flatMap((address: string) => {
-                return [
-                    [address, ORACLE_TEMPLATE_FUNCTION_DATA],
-                    [address, ORACLE_FINALIZED_FUNCTION_DATA],
-                ];
-            })
-        );
+        const {
+            result: [, oraclesResult],
+        } = await publicClient.simulateContract({
```

Comment:
Why are we not using `publicClient.multicall` for the multicalls?

Reply:
After trying, this is actually a bit of a mess with types. Let's do this at a later time. I've added a TODO here: 16d04b5e7c3792d0315ffb08f8792c9585447966

Reply:
Oh ok, let's keep `simulateContract` then.